service/http/topic.go

```diff
@@ -0,0 +1,66 @@
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+func (s *ServiceImp) registerSubscribeHandler() {
+	f := func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		if err := json.NewEncoder(w).Encode(s.topicSubscriptions); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+	s.mux.HandleFunc("/dapr/subscribe", f)
+}
+
+// AddTopicEventHandler appends provided event handler with it's name to the service
+func (s *ServiceImp) AddTopicEventHandler(topic, route string, fn func(ctx context.Context, e *TopicEvent) error) error {
+	if topic == "" {
+		return errors.New("topic name required")
+	}
+
+	if route == "" {
+		return errors.New("handler route name")
+	}
+
+	sub := &Subscription{Topic: topic, Route: route}
```

Comment:
same here, I would allow to pass metadata to the topic list

Reply:
same as above, HTTP serving implementation was removed now.

Reply:
http serving has been removed per common SDK design decision